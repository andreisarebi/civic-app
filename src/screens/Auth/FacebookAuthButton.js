import React from 'react';
import { StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import SocialButton from './SocialButton';
import colors from '../../styles/colors';
import { facebookLogin } from '../../auth/redux/userReducer';
import { getIsLoading } from '../../auth/redux/selectors';

const FacebookAuthButton = (props) => {
  const textColor = colors.white;
  const icon = { name: 'facebook-box', type: 'material-community', size: 35 };
  return (
    <SocialButton
      textColor={textColor}
      icon={icon}
      style={props.style}
      buttonStyle={styles.button}
      onPress={props.onPress}
      title={props.title}
    />
  );
};

FacebookAuthButton.propTypes = {
  onPress: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
  style: SocialButton.propTypes.style,
};

FacebookAuthButton.defaultProps = { style: {} };

const styles = StyleSheet.create({
  button: {
    backgroundColor: colors.facebookBlue,
    borderColor: colors.facebookBlue,
  },
});

const mapStateToProps = state => ({ isLoading: getIsLoading(state) });

const mapDispatchToProps = { onPress: facebookLogin };

// Prevent monkey clicking the button by disabling the onPress
// method if there's already an in-flight request
const mergeProps = (stateProps, dispatchProps, ownProps) => ({
  onPress: () => !stateProps.isLoading && dispatchProps.onPress(),
  ...ownProps,
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
  mergeProps,
)(FacebookAuthButton);
