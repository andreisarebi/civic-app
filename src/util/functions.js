import distanceInWordsStrict from 'date-fns/distance_in_words_strict';

export const distanceInWordsFromNow = (date) => {
  return distanceInWordsStrict(date, Date.now());
}
