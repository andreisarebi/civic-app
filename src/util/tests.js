import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { Provider } from 'react-redux';
import store from '../../App/configureStore';

export const createSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Component {...props} />);
  return renderer.getRenderOutput();
};

export const createConnectedSnapshot = (Component, props) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Provider store={store}><Component {...props} /></Provider>);
  return renderer.getRenderOutput();
};
