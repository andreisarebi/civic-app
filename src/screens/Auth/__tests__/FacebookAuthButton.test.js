import { createSnapshot } from '../../../util/tests';
import FacebookAuthButton from '../GoogleAuthButton';

describe('FacebookAuthButton', () => {
  it('renders correctly', () => {
    const snapshot = createSnapshot(FacebookAuthButton.WrappedComponent, {
      title: 'continue with facebook',
      onPress: jest.fn(),
    });
    expect(snapshot).toMatchSnapshot();
  });
});
