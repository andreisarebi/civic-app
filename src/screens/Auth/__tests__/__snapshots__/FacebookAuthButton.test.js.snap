// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookAuthButton renders correctly 1`] = `
<SocialButton
  buttonStyle={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#e0e0e0",
    }
  }
  icon={Object {}}
  iconComponent={[Function]}
  onPress={[MockFunction]}
  style={Object {}}
  textColor="#000000"
  title="continue with facebook"
/>
`;
